components/base/wheeled/wheeled_base.go

```diff
@@ -471,24 +464,57 @@ func (wb *wheeledBase) straightDistanceToMotorInputs(distanceMm int, mmPerSec fl
 
 // Stop commands the base to stop moving.
 func (wb *wheeledBase) Stop(ctx context.Context, extra map[string]interface{}) error {
-	var err error
-	for _, m := range wb.allMotors {
-		err = multierr.Combine(err, m.Stop(ctx, extra))
+	stopFuncs := func() []rdkutils.SimpleFunc {
+		ret := []rdkutils.SimpleFunc{}
+
+		wb.mu.Lock()
+		defer wb.mu.Unlock()
+		for _, m := range wb.left {
+			motor := m
+			ret = append(ret, func(ctx context.Context) error { return motor.Stop(ctx, extra) })
+		}
+
+		for _, m := range wb.right {
+			motor := m
+			ret = append(ret, func(ctx context.Context) error { return motor.Stop(ctx, extra) })
+		}
+		return ret
+	}()
+
+	if _, err := rdkutils.RunInParallel(ctx, stopFuncs); err != nil {
+		return multierr.Combine(err, wb.Stop(ctx, nil))
 	}
-	return err
+	return nil
 }
 
 func (wb *wheeledBase) IsMoving(ctx context.Context) (bool, error) {
-	for _, m := range wb.allMotors {
-		isMoving, _, err := m.IsPowered(ctx, nil)
-		if err != nil {
-			return false, err
+	moveFuncs := func() []rdkutils.ReturnFunc[bool] {
+		ret := []rdkutils.ReturnFunc[bool]{}
+
+		wb.mu.Lock()
+		defer wb.mu.Unlock()
+		for _, m := range wb.left {
+			motor := m
+			ret = append(ret, func(ctx context.Context) (bool, error) { return motor.IsMoving(ctx) })
 		}
-		if isMoving {
-			return true, err
+
+		for _, m := range wb.right {
+			motor := m
+			ret = append(ret, func(ctx context.Context) (bool, error) { return motor.IsMoving(ctx) })
 		}
+		return ret
+	}()
+
+	_, bools, err := rdkutils.GetInParallel(ctx, moveFuncs)
+	if err != nil {
+		return false, multierr.Combine(err, wb.Stop(ctx, nil))
+	}
+
+	result := false
+	for _, b := range bools {
+		result = result || b
```

Comment:
now using generic with `GetInParallel` which is nice but it still feels like an unnecessary step. @randhid or @oliviamiller any thoughts on why this is better or if it should just loop through regularly?

Reply:
it's less code to just loop through regularly so maybe its better to do that? But not sure if doing it in parallel has other advantages that i'm unaware of so don't have a strong opinion.